### PR TITLE
new headwear and headwear sprite edits

### PR DESCRIPTION
**Pull request by nefarious6th**
[qol][wiki][cleanliness]
## About the PR 

**Added Clothes**
- adds some more headbands; they can be found in a cargo crate and potentially as trinkets. added variants are: leopard ears, snow leopard ears, tiger ears, bee antennae, and giraffe ears.
ears
- adds barrettes; can be found in the clothing booth. added variants are: blue butterfly, orange butterfly, blue, green, pink, and gold. red hair clips available in the clothing booth now too.
clips

**Back-End Things To Do With Headbands**
- makes an in-hand icon for headbands
- repaths all headbands to be under `/obj/item/clothing/head/headband`, and allows for all of the child objects of the type to have radio headsets attached to them and have them worn as ear-slot items
- makes wannaBEEs start the round wearing an antennae headband, to enhance bee-lief that they are actually bees
- blacklists barrettes added in this pr from the trinket hat list so as not to overly skew the probability of people getting their Epic Gamer Syndie Helmets or whatever

**Krimpus? Krimpus.**
- makes krimpus start the round wearing the botanist's apron because it is cute and a good sprite

**Really Exciting And Cool Stuff Everyone Surely Wants To Read About**
- cleans up some stuff on the yeti earmuff worn sprites
- fixes sprite errors with the purple catears
- resprites the witchfinder hat item sprite to look more like a capotain

## Why's this needed?
*I* need this, this pr's primary audience is Me.
Barrettes were requested as a feature in Discord.
customization is fun. i thought the monkey apron thing and the bee headband give some fun goon flavor to things already going on in the codebase. what is a bee without their antennae, after all?

```changelog
(u)nefarious6th
(*)Added numerous new headwear clothing options to the clothing booth, new clothing qm crate, general sprite fixes.
```

### CODE REVIEW

**Comment by MatheusIsNotAvailable** [comment]
Afaik, syndie helmets are already blocked from the hat trait so it is probably fine? I am personally not against people getting giraffe headbands instead of some helmet

Also, and this is more of a suggestion for a future thing, you should look into making greyscale cat ears and barrettes just like the colorful beret, might be cool.

File: code/procs/jobprocs.dm [hunk 1]
```diff
@@ -556,7 +556,7 @@
 	else if (src.traitHolder && src.traitHolder.hasTrait("loyalist"))
 		trinket = new/obj/item/clothing/head/NTberet(src)
 	else if (src.traitHolder && src.traitHolder.hasTrait("petasusaphilic"))
-		var/picked = pick(childrentypesof(/obj/item/clothing/head) - (typesof(/obj/item/clothing/head/bighat)) - (typesof(/obj/item/clothing/head/helmet/space/syndicate))) //IM A MONSTER DONT LOOK AT ME. NOOOOOOOOOOO
+		var/picked = pick(childrentypesof(/obj/item/clothing/head) - (typesof(/obj/item/clothing/head/barrette/)) - (typesof(/obj/item/clothing/head/headband/bee)) - (typesof(/obj/item/clothing/head/headband/giraffe)) - (typesof(/obj/item/clothing/head/headband/nyan/leopard)) - (typesof(/obj/item/clothing/head/headband/nyan/snowleopard)) - (typesof(/obj/item/clothing/head/headband/nyan/tiger)) - (typesof(/obj/item/clothing/head/bighat)) - (typesof(/obj/item/clothing/head/helmet/space/syndicate))) //IM A MONSTER DONT LOOK AT ME. NOOOOOOOOOOO
```
Comment (Sovexe) [inline code/procs/jobprocs.dm:559]: maybe just make this use a var or something at this point

File: code/modules/vending/vending.dm [hunk 2]
```diff
@@ -2590,15 +2590,17 @@
...
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/white, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/gray, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/black, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/red, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/orange, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/yellow, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/green, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/blue, 3, cost = PAY_TRADESMAN)
-		product_list += new/datum/data/vending_product(/obj/item/clothing/head/nyan/purple, 3, cost = PAY_TRADESMAN)
+		product_list += new/datum/data/vending_product(/obj/item/clothing/head/headband/nyan/white, 3, cost = PAY_TRADESMAN)
```
Comment (Sovexe) [inline code/modules/vending/vending.dm:2593]: seems kind of weird to have the same item available in two different vendors, and at two different prices

maybe just consolidate to the booth

**Review by Sovexe — CHANGES_REQUESTED** [review]

**Review by flrsh — CHANGES_REQUESTED** [review]
Cute hairclips. Instead of the clothing booth, could you put the gimmick (leopard, snow leopard, tiger, giraffe, bee, antler) headbands in a QM headband crate?

File: code/procs/jobprocs.dm (hunk 1 above)
Comment (nefarious6th) [reply code/procs/jobprocs.dm]: let me know if what i implemented seems okay? i tried to just follow the trinket blacklist format but i could also try just adding a var onto hats we want blacklisted?

File: code/procs/jobprocs.dm [hunk 3]
```diff
@@ -556,7 +556,7 @@
 	else if (src.traitHolder && src.traitHolder.hasTrait("loyalist"))
 		trinket = new/obj/item/clothing/head/NTberet(src)
 	else if (src.traitHolder && src.traitHolder.hasTrait("petasusaphilic"))
-		var/picked = pick(childrentypesof(/obj/item/clothing/head) - (typesof(/obj/item/clothing/head/bighat)) - (typesof(/obj/item/clothing/head/helmet/space/syndicate))) //IM A MONSTER DONT LOOK AT ME. NOOOOOOOOOOO
+		var/picked = pick(!hat_blacklist)
```
Comment (Sovexe) [inline code/procs/jobprocs.dm:559]: pick from the negation of a list of types?

Comment (nefarious6th) [reply code/procs/jobprocs.dm:559]: does `var/picked = pick(childrentypesof(/obj/item/clothing/head) - (hat_blacklist))` seem right? to use instead?

Comment (ZeWaka) [reply code/procs/jobprocs.dm]: Why is not concretetypesof and the clothing/head abstract

File: code/procs/jobprocs.dm [hunk 4]
```diff
@@ -781,3 +781,10 @@ var/list/trinket_safelist = list(/obj/item/basketball,/obj/item/instrument/bikeh
 /obj/item/toy/plush/small/bee, /obj/item/paper/book/from_file/the_trial, /obj/item/paper/book/from_file/deep_blue_sea, /obj/item/clothing/suit/bedsheet/cape/red, /obj/item/disk/data/cartridge/clown,
 /obj/item/clothing/mask/cigarette/cigar, /obj/item/device/light/sparkler, /obj/item/toy/sponge_capsule, /obj/item/reagent_containers/food/snacks/plant/pear, /obj/item/reagent_containers/food/snacks/donkpocket/honk/warm,
 /obj/item/seed/alien)
+
+
+
+////////////////////////////////////
+// hat blacklist for trinket hats //
+////////////////////////////////////
+var/list/hat_blacklist = list((typesof(/obj/item/clothing/head/bighat)), (typesof(/obj/item/clothing/head/helmet/space/syndicate)), (typesof(/obj/item/clothing/head/barrette)))
```
Comment (Sovexe) [inline code/procs/jobprocs.dm:790]: only a single typesof call is needed here
```suggestion
var/list/hat_blacklist = list(typesof(/obj/item/clothing/head/bighat, /obj/item/clothing/head/helmet/space/syndicate, /obj/item/clothing/head/barrette))
```

File: code/procs/jobprocs.dm [hunk 5]
```diff
@@ -600,7 +600,7 @@
 	else if (src.traitHolder && src.traitHolder.hasTrait("loyalist"))
 		trinket = new/obj/item/clothing/head/NTberet(src)
 	else if (src.traitHolder && src.traitHolder.hasTrait("petasusaphilic"))
-		var/picked = pick(childrentypesof(/obj/item/clothing/head) - (typesof(/obj/item/clothing/head/bighat)) - (typesof(/obj/item/clothing/head/helmet/space/syndicate))) //IM A MONSTER DONT LOOK AT ME. NOOOOOOOOOOO
+		var/picked = pick(childrentypesof(/obj/item/clothing/head) - (hat_blacklist))
```
Comment (Sovexe) [inline code/procs/jobprocs.dm:603]: concrete types are preferable
```suggestion
		var/picked = pick(concrete_typesof(/obj/item/clothing/head) - hat_blacklist)
```

**Review by Sovexe — APPROVED** [review]

**Pull request merged**